python/lsst/pipe/base/pipelineTask.py

```diff
@@ -244,22 +244,77 @@ def getDatasetTypes(cls, config, configClass):
                 dsTypes[key] = cls.makeDatasetType(value)
         return dsTypes
 
-    def run(self, *args, **kwargs):
+    def runWithOutputs(self, inputData, outputDataIds):
         """Run task algorithm on in-memory data.
 
-        This function is the one that actually operates on the data and usually
-        returning a `Struct` with the produced results. This method will be
-        overridden by every subclass. It operates on in-memory data structures
-        (or data proxies) and cannot access any external data such as data
-        butler or databases. All interaction with external data happens in
-        `runQuantum` method.
-
-        With default implementation of `runQuantum()` this method will
-        receive keyword arguments whose names will be the same as names
-        of configuration fields describing input and output dataset types.
-        For input dataset types argument values will be lists of the data
-        object retrieved from data butler. For output dataset types argument
-        values will be the lists of units from DataRefs in a Quantum.
+        This method is called by :py:meth:`runQuantum` to operate on input
```

Comment:
You should just be able to say `runQuantum` without the `:py:meth:` business; I'm not sure where the latter is required (@jonathansick?) but I'm pretty sure links in docstrings will work fine without it as long as there are no namespace-qualification issues (and there shouldn't be for a method of the same class).

(Several other instances of this throughout the changeset that should also be fixed.)

Reply:
I think that is correct, only the backticks should be needed